scrapy_autoextract/providers.py

```diff
@@ -14,6 +16,10 @@
     register,
 )
 
+AUTOEXTRACT_EXTRA_KEY = "__autoextract_extra"
+AUTOEXTRACT_PROVIDERS_KEY = "__autoextract_providers"
+AUTOEXTRACT_RESPONSE_KEY = "__autoextract_response"
```

Comment:
Some data is stored on Scrapy's requests. 

- it works even with async providers because we're mainly using it
  - in sync before hooks
  - writing in a single coroutine
  - reading in multiple coroutines

Alternatively, we could use thread-local storage indexed by the request's fingerprint instead of using the request directly to avoid issues with disk queues.